### DISCUSSION TITLE
Where '-' hyphen minus 0x2D conversion to soft hyphen (0xAD) is done.

### DISCUSSION THREAD

**Discussion by PPakalns** [typst/typst#2630] [Q&A] [answered]
Related to issue: https://github.com/typst/typst/issues/2628

I want to create temporary typst fork with hyphen character conversion disabled.

I tried to search all codebase about how '-' is handled.  What i found out is that it is handled probably as Shorthand, but i couldn't find out where the character is itself replaced with soft hyphen. Is it even done inside this repository?

https://github.com/search?q=repo%3Atypst%2Ftypst%2000AD&type=code   (This converts only -? soft hyphen shorthand not minus hyphen. 
https://github.com/search?q=repo%3Atypst%2Ftypst+%27-%27&type=code

There are definitions for math minus conversion, but in reality in raw block hyphen is not converted to math minus, but to soft hyphen.
```
("-", '\u{2212}')
```

Maybe someone can point me in the right direction where this conversation '-' (0x2D) to soft hyphen (0xAD) happens.

Thank you for your help.

**Comment by laurmaedje** [comment, marked as answer]
There is no such conversion in Typst. The conversion happens in your PDF reader because it has "smart" behaviour to get rid of hyphenation artifacts. There is no easy workaround until Typst implements the special behaviour mandated by the PDF spec as discussed in https://github.com/typst/typst/issues/1267.

**Reply by PPakalns** [reply]
Found another workaround, with other fonts the problem didn't happen :tada:
